### ISSUE TITLE
If custom_field name is "id", "ID", "iD", etc., article's SQL query can fail

### ISSUE THREAD

**Issue by kuopassa** [textpattern/textpattern#1624]
### Steps to reproduce

Found with tired eyes, so maybe I missed something... :-) If Textpattern's custom_field contains the magical word `id`, then at least this tag will likely fail to find the right article:

`<txp:article_custom id="4" status="5" form="default" />`

That's because the SQL query looks like (according to Textpattern's debugging mode):

`SELECT *, UNIX_TIMESTAMP(Posted) AS uPosted, UNIX_TIMESTAMP(Expires) AS uExpires, UNIX_TIMESTAMP(LastMod) AS uLastMod FROM textpattern WHERE 1 AND `Posted` <= from_unixtime(2147483647) AND (Expires IS NULL OR from_unixtime(2147483647) <= Expires) AND ID  IN (4) AND Status IN (5) AND (custom_3 LIKE '4')  ORDER BY FIELD(ID, 4), Posted DESC LIMIT 0, 10`

The interesting part is `AND (custom_3 LIKE '4')` because instead of `custom_3` that should be `ID`. `custom_3`'s name is `id`.

#### Additional information

Textpattern version: Textpattern 4.8.4

Web server vendor and version: Apache 2.4

Database server vendor and version: MariaDB 10

PHP version: 7.3

**Comment by Bloke** [comment]
Yes. We should add 'id' to the list of [words that are not a good idea to use as custom field names](https://docs.textpattern.com/administration/preferences-panel#custom-fields-preferences) and ensure that list is transferred over to the new documentation when we move things around.

**Comment by bloatware** [comment]
It actually was in `any other tag attribute, just in case!` But there are also `description`, `expires`, `modified`, `override_form`, `is_first` and `is_last` to add to the list.

**Comment by bloatware** [comment]
And I don't remember why we don't internally store them as `$thisarticle['custom_1']` to avoid such clashes.

**Comment by Bloke** [comment]
We will (do!) store them that way in the new custom fields implementation, for backwards compatibility with plugins. They are also populated as 'namespaced' values with a prefix (`txp_cf_` if memory serves) so that they won't clash with any internals any more. 

I hit an issue today because I called one of my custom fields 'Class' and it killed the `<txp:article>' tag when I passed wraptag/class attributes in. **sigh**
